### ISSUE TITLE
Add Gitalk Comment | lanVoyager | toHave-N-haveNot

### ISSUE THREAD

**Issue by lanVoyager** [lanVoyager/gitalkHexoComment#5]
https://lanvoyager.github.io/posts/66a6d538/ 

 The Comment Plug-InToday is my day 7 with Hexo. Goal is to add comment feature here.  My requirements:  Simple and elegant Easy to manage Allow Anonymous comments FREE  As always, let’s begin with som

**Comment by marlluslustosa** [comment]
Have you had any success using adapting gitalk for anonymous comments?
If you want to join the discussion, we are here: https://github.com/utterance/utterances/issues/542
and https://github.com/gitalk/gitalk/issues/455

**Comment by lanVoyager** [comment]
> @marlluslustosa
> Have you had any success using adapting gitalk for anonymous comments?
> If you want to join the discussion, we are here: https://github.com/utterance/utterances/issues/542
> and https://github.com/gitalk/gitalk/issues/455

Thank you for reading. No, I have not resolved the anonymous comment issue yet.

**Comment by chinsengi** [comment]
> @marlluslustosa
> Have you had any success using adapting gitalk for anonymous comments?
> If you want to join the discussion, we are here: https://github.com/utterance/utterances/issues/542
> and https://github.com/gitalk/gitalk/issues/455

Yes, that would be such a great feature. Making gittalk the winner of all

**Comment by marlluslustosa** [comment]
I think so.
In my tests I managed to get this to work, but I think it would need an extra effort, which is the user to host an intermediate server (vercel, cloudflare workers...) to host the code that uses the bot's Personal Access Token (PAT) who will comment anonymously. This is solve secret and password leaking issues put on the frontend of site.

But it's completely workable. I'm preparing a post about it.
I will soon release an update and post it here.

**Comment by marlluslustosa** [comment]
> > @marlluslustosa
> > Have you had any success using adapting gitalk for anonymous comments?
> > If you want to join the discussion, we are here: [utterance/utterances#542](https://github.com/utterance/utterances/issues/542)
> > and [gitalk/gitalk#455](https://github.com/gitalk/gitalk/issues/455)
> 
> Yes, that would be such a great feature. Making gittalk the winner of all

I think so.
In my tests I managed to get this to work, but I think it would need an extra effort, which is the user to host an intermediate server (vercel, cloudflare workers...) to host the code that uses the bot's Personal Access Token (PAT) who will comment anonymously. This is solve secret and password leaking issues put on the frontend of site.

But it's completely workable. I'm preparing a post about it.
I will soon release an update and post it here.

**Comment by marlluslustosa** [comment]
I posted a solution for this on my blog (in Portuguese). It's not translated into English yet, so use a translator.
https://marllus.com/tecnologia/2021/06/23/gitalk-comentarios-anon
